service/listenerpipe.go

```diff
@@ -0,0 +1,53 @@
+package service
+
+import (
+	"errors"
+	"net"
+	"sync"
+)
+
+// ListenerPipe, like net.Pipe, returns a full-duplex in-memory connection.
+// One end of the connection is returned as a net.Conn interface.
+// The other end is returned as a listener that will return it  when Accept
```

Comment:
nit: This comment needs a little rewording.

Reply:
Is this better or do you have something else in mind?